src/CoreWCF.BuildTools/src/OperationParameterInjectionGenerator.Emitter.cs

```diff
@@ -141,6 +141,15 @@ namespace {operationContractSpec.ServiceContractImplementation.ContainingNamespa
                 indentor.Increment();
                 builder.AppendLine($@"{indentor}var serviceProvider = CoreWCF.OperationContext.Current.InstanceContext.Extensions.Find<IServiceProvider>();");
                 builder.AppendLine($@"{indentor}if (serviceProvider == null) throw new InvalidOperationException(""Missing IServiceProvider in InstanceContext extensions"");");
+
+                if (dependencies.Any(x => SymbolEqualityComparer.Default.Equals(x.Type, operationContractSpec.HttpContextSymbol)
+                    || SymbolEqualityComparer.Default.Equals(x.Type, operationContractSpec.HttpRequestSymbol)
+                    || SymbolEqualityComparer.Default.Equals(x.Type, operationContractSpec.HttpResponseSymbol)))
+                {
+                    builder.AppendLine($@"{indentor}var httpContext = CoreWCF.OperationContext.Current.RequestContext.RequestMessage.Properties[""Microsoft.AspNetCore.Http.HttpContext""] as Microsoft.AspNetCore.Http.HttpContext;");
+                    builder.AppendLine($@"{indentor}if (httpContext == null) throw new InvalidOperationException(""Missing HttpContext in RequestMessage properties"");");
```

Comment:
The this[string] item accessor for MessageProperties throws ArgumentException if the item isn't found so your exception won't get thrown. Maybe use ContainsKey to throw. Or you could do the check and fetch with TryGetValue. The returned value may still be null if there's a key with that name but the value is null.

Reply:
I fixed the generated code in latest commit